lib/Validator.js

```diff
@@ -343,6 +340,11 @@ class Validator {
           const results = anyOfValidators.map((fn) => fn(value));
           const valid = results.some((result) => result === undefined);
           if (!valid) {
+            // If every condition gives format error, just return one error
+            // TODO does this apply in allOf, second branch of exactlyOneOf?
+            if (results.every((result) => result === parse_error)) {
+              return parse_error;
+            }
             return results.join('\n');
```

Comment:
Test:
`expect(fn('hello')).toEqual('Value does not match format for xsd:integer')`

We were getting `'Value does not match format for xsd:integer\Value does not match format for xsd:integer'` without this new `if` condition.

Was the test wrong? Do want duplicated format error messages across two `anyOf` conditions? If so, I would remove this `if` block and edit the test.

Or do we want to keep this `if` block, and add a similar one for `allOf`, and the second return branch of `exactlyOneOf`?

Reply:
Keep two error messages, but let's make them more descriptive. Which anyOf conditions are they violating?